Documentation/ApiOverview/Yaml/Index.rst

```diff
@@ -56,3 +56,64 @@ Examples:
      - { resource: "EXT:rte_ckeditor/Configuration/RTE/Processing.yaml" }
      - { resource: "misc/my_options.yaml" }
      - { resource: "../path/to/something/within/the/project-folder/generic.yaml" }
+
+
+Custom Placeholder Processing
+-----------------------------
+
+It is possible to register custom placeholder processors to allow fetching data from
+different sources. To do so, register a custom processor via :file:`LocalConfiguration.php`:
+
+.. code-block:: php
+
+   $GLOBALS['TYPO3_CONF_VARS']['SYS']['yamlLoader']['placeholderProcessors'][\Vendor\MyExtension\PlaceholderProcessor\CustomPlaceholderProcessor::class] = [];
+
+There are some options available to sort or disable placeholder processors if necessary:
+
+.. code-block:: php
+
+   $GLOBALS['TYPO3_CONF_VARS']['SYS']['yamlLoader']['placeholderProcessors'][\Vendor\MyExtension\PlaceholderProcessor\CustomPlaceholderProcessor::class] = [
+      'before' => [
+         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\ValueFromReferenceArrayProcessor::class
+      ],
+      'after' => [
+         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\EnvVariableProcessor::class
+      ],
+      'disabled' => false
+   ];
+
+New placeholder processors must implement the :php:`\TYPO3\CMS\Core\Configuration\Processor\Placeholder\PlaceholderProcessorInterface`
+An implementation may look like the following:
+
+.. code-block:: php
+
+   class ExamplePlaceholderProcessor implements PlaceholderProcessorInterface
+   {
+      public function canProcess(string $placeholder, array $referenceArray): bool
+      {
+         return strpos($placeholder, '%example(') !== false;
+      }
+
+      public function process(string $value, array $referenceArray)
+      {
+         // do some processing
+         $result = $this->getValue($value);
```

Comment:
We should add at least a simulated method for this.

Reply:
tried to add one, it doesn't make much sense but I couldn't think of a better one.